aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -170,11 +171,12 @@ public void write(byte[] b, int off, int len) throws IOException {
     }
   }
 
-  private void newStream() throws IOException {
+  private void newStream() throws IOException, SecurityException {
```

Comment:
Sorry for the back and forth, I read the documentation for mkdirs, it seems like SecurityException only catches JVM level permission and it might still just return false for OS level permission failure, so it's hard to have a consistent behavior for error handling. Because of that, plus the fact that SecurityException is a runtime exception, I think we can remove the special handling of it and just let it throw to the top level.

Reply:
Removed the special handling for `SecurityException`.

Reply:
Yeah, need not to throw explicitly. Removed it.